### PR TITLE
Add development workflow documentation

### PR DESCRIPTION
**Pull request by cutwater**

### CODE REVIEW

**Comment by chouseknecht** [comment]
@cutwater 

If we're going to have a `development` section in docs, then we need to do the following:

- move the contributing guide to the `development` section
- change README to link to the `development` section for both contributing and branch strategy

File: docs/docsite/rst/developers/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+    galaxy-dev          latest              2f552729e204        22 seconds ago      748MB
+    galaxy-build        latest              c456f5c226d3        2 minutes ago       568MB
+    centos              7                   196e0ce0c9fb        6 weeks ago         197MB
+
+Start the containers
+--------------------
+
+We use ``docker-compose`` to run the containers. If you're curious about 
+the services, and and how they're configured, the compose file is 
+[scripts/docker/dev/compose.yml](./scripts/docker/dev/compose.yml)
```
Comment (chouseknecht) [inline docs/docsite/rst/developers/contributing.rst]: Link should be in RST format

File: docs/docsite/rst/developers/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+    ## Set your working directory to the project root
+    $ cd galaxy
+
+    ## Start the build process
+    $ make dev/up_detached
+
+
+Since the ``stdout`` and ``stderr`` are not streaming to your terminal 
+session, you'll need to use the ``docker logs`` command to view logging output.
+As pictured above in [quick start](#quick-start), use ``docker ps`` to see
```
Comment (chouseknecht) [inline docs/docsite/rst/developers/contributing.rst]: Link should be in RST format

File: docs/docsite/rst/developers/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+     UI External: http://172.18.0.6:3001
+     -----------------------------------
+    [Browsersync] Serving files from: /galaxy
+    [03:10:04] Finished 'less' after 3.23 s
+    [03:10:04] Starting 'default'...
+    [03:10:04] Finished 'default' after 108 μs
+
+OK, go for it! Your Galaxy web site is available at: http://localhost:8000.
+
+.. note:: You won't be able to authenticate until you perform the post build
```
Comment (chouseknecht) [inline docs/docsite/rst/developers/contributing.rst]: Might be wrong, but think there should be a line break after `.. note::`

File: docs/docsite/rst/developers/contributing.rst [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+
+* Set the **Provider** to ``GitHub``.
+* Enter ``GitHub`` as the **Name**.
+* From the new GitHub OAuth application you just created, copy the 
+  **ClientID** value into **Client id**.
+* Copy the **Client Secret** value into **Secret key**.
+* Under **Sites**, add ``localhost`` to **Chosen sites**. Save the changes.
+
+Now test the authentication. Log out of your admin account, and go back to the 
+home page at http://localhost:8000. Now log in using your GitHub account by
```
Comment (chouseknecht) [inline docs/docsite/rst/developers/contributing.rst]: Highlight the URL with double ticks, to be consistent with other links.

File: docs/docsite/rst/developers/contributing.rst [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+
+Static HTML files may be used simply by opening them in your favourite web browser.
+
+Commiting
+---------
+
+Once you have all lint and tests passed and you are ready to commit your patch 
+and propose pull request, install pre-commit hook, provided by this repository::
+
+    $ cp pre-commit .git/hooks/pre-commit
```
Comment (chouseknecht) [inline docs/docsite/rst/developers/contributing.rst]: Missing `.. code-block::`

File: docs/docsite/rst/developers/workflow.rst [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+.. note: Check cherry-picker tool to automate backporting process.
+
+.. _release_process:
+
+Release process
+===============
+
+Merge ``release/<x.x.x>`` branch to master::
+
+     git checkout master
```
Comment (chouseknecht) [inline docs/docsite/rst/developers/workflow.rst]: Should this be a `.. code-block::`?

File: docs/docsite/rst/developers/contributing.rst (hunk 3 above)
Comment (cutwater) [reply docs/docsite/rst/developers/contributing.rst]: Nope, there is no break required.
http://docutils.sourceforge.net/docs/ref/rst/directives.html#specific-admonitions

File: docs/docsite/rst/developers/contributing.rst (hunk 4 above)
Comment (cutwater) [reply docs/docsite/rst/developers/contributing.rst]: Done.

File: docs/docsite/rst/developers/workflow.rst [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+    git push origin backport/3.0/[PR_NUMBER_FROM_DEVEL]
+
+
+Submit the pull request for ``backport/3.0/[PR_NUMBER_FROM_DEVEL]`` against
+the ``release/3.0`` branch.
+
+Name your pull request accordingly to the original one. 
+Include a reference to the original pull request into description 
+in format (. Assign a ``backport`` label.
```
Comment (webknjaz) [inline docs/docsite/rst/developers/workflow.rst]: > ` (. `

@cutwater ^^^

**Pull request merged**